### ISSUE TITLE
[REQ] Margin trading feature

### ISSUE THREAD

**Issue by Gabri** [SockTrader/SockTrader#14]
Hello

I think a very nice feature could be Margin trading support (for now just backtesting), so not just buy and sell, but long, short and leverage.
An initial implementation could simply transform a current sell in a closing eventual long position and open a short one.

Thanks

**Comment by cwouter** [comment]
Hi @Gabri ,

Thanks for your feature request. To be honest I don't have enough experience with margin trading at the moment. I know that it works by selling coins that you've lend from someone else and buying back in at a lower rate (hopefully). But I guess that you would also need to pay an interest fee for each day that you have an open position? How would you determine the amount of daily interest?

And what about getting "margin called", how would you see this and make it as realistic as possible in a backtest?

I will create a new ticket once we thoroughly discussed every detail and have a clear description with acceptance criteria.

**Comment by Gabri** [comment]
Hi,

Yes, it's not so simple if you want to calculate profit/loss to cents.
Usually it's applied a funding rate to the open positions, but this funding could be positive or negative depends on number of shorts and longs present in that moment (in perpetual contracts, for example, Bitmex calculate it every 8 hours) and if you close the position prior to the funding exchange then you will not pay or receive funding. But different exchanges can have different behaviors.
Depends on the type of order there will be a taker or maker fees (as every exchanges).
This kind of trading is quite popular because you can win in every market direction, in addition there is the leverage that can multiply your profit/loss.
In tradingview too, when you develop a strategy in pine script (I always develop there a strategy), it calculates profit/loss as buy signals are longs and sell signals are short (so, if you a have a sell signal and you had a previous long position, it close it before open the short one. Same things viceversa with buy and previous short position. Of course it's also possibile to close a position with different parameters not just if you trigger a sell/buy signal).

Probably I'm wrong but maybe funding fees are not so radically important.
Does SockTrader now support fee calculation for maker/taker trades? When a buy call is done is intended as market order?

Thnaks

**Comment by cwouter** [comment]
I'll close this issue for now. It is definitely on our roadmap but we first need an exchange with margin trading before we can start thinking about this improvement.
